src/backends/perf_pt/collect.c

```diff
@@ -246,6 +368,10 @@ poll_loop(int perf_fd, int stop_fd, struct perf_event_mmap_page *mmap_hdr,
     }
 
 done:
+    if (data_tmp != NULL) {
```

Comment:
I think `data_type` is always `!= NULL` at this point.

Reply:
If allocation fails, `data_tmp` is `NULL`.